Add release notes for 2.0.1 (#6123)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## 2.0.1

> The response Prefect 2 has been overwhelming in the best way possible! Thank you to the many users who alerted us to minor issues. This release is a compilation of enhancements abnd fixes that make for a more resilient, performant, and refined Prefect experience. Thanks in particular to the students at this week's Prefect Associate Certification Course (PACC) in San Jose, for their thoughtful feedback.

Author: ```suggestion
The response to Prefect 2 has been overwhelming in the best way possible! Thank you to the many users who alerted us to minor issues. This release is a compilation of enhancements and fixes that make for a more resilient, performant, and refined Prefect experience. Thanks in particular to the students at this week's Prefect Associate Certification Course (PACC) in San Jose, for their thoughtful feedback.
```

> The response to Prefect 2 has been overwhelming in the best way possible! Thank you to the many users who alerted us to minor issues. This release is a compilation of enhancements and fixes that make for a more resilient, performant, and refined Prefect experience. Thanks in particular to the students at this week's Prefect Associate Certification Course (PACC) in San Jose, for their thoughtful feedback.

Reviewer: ```suggestion
The response to Prefect 2 has been overwhelming in the best way possible. Thank you to the many community members who tried it out and gave us feedback! Thanks in particular to the students at this week's Prefect Associate Certification Course (PACC) in San Jose, for their thoughtful recommendations. This release is a compilation of enhancements and fixes that make for a more resilient, performant, and refined Prefect experience.
```

### Improvements and bug fixes
- Schedules set via the API or UI are now preserved when building deployments from the CLI
- JSON types are now coerced to none, following Javascript convention and supporting standards compatibility

> - The `prefect deployment execute` command has been removed

Reviewer 2: ```suggestion
- The `prefect deployment execute` command has been removed to avoid confusion between running a flow locally from a Python script and running it by an agent using `prefect deployment run`
```

> - This repository now includes templates for pull requests and issues

Reviewer 2: ```suggestion
- This repository now includes templates for pull requests and issues to make bug reports and community contributions easier
```

> - A common request we immediately received was for unique deployment YAML names (to better anticipate multi-flow deployments) and an optional output flag to customize the name.

Reviewer 2: ```suggestion
- A common request we immediately received was for unique deployment YAML names (to better anticipate multi-flow deployments) and an optional output flag to customize the name. As a result, the `deployment build` command now supports an optional output flag to customize the name of the deployment.yaml file and thus, to better support projects with multiple flows
```

Reviewer: I'm appending this to the other bullet

- The `scheduler` and `flow-run-notifications` LoopServices have been made more resilient
- Log inserts have been made more performant through smaller log batches

> - Local file system blocks can no longer be created from the UI

Reviewer 2: ```suggestion
- Local file system blocks created from the UI now point to the right `base_path`
```

sorry this one is on me - I should have described my PR better 🙏

> - Support for unmapped values to Task.map has been added

Reviewer 2: ```suggestion
- Support for unmapped values to Task.map has been added as requested by Club42 members
```

Reviewer 2: Feel free to skip this one but I thought this might be a cool note to show that we actually listen to their requests

Reviewer: Definitely!

> - The `deployment build` command now supports an optional output flag to customize the name of the deployment.yaml file

Reviewer 2: ```suggestion
```

covered earlier - redundant

Reviewer: Thank you
